plugins/plugin-chart-pivot-table/src/PivotTableChart.tsx

```diff
@@ -118,16 +90,92 @@ export default function PivotTableChart(props: PivotTableProps) {
       ...acc,
       ...metricNames.map((name: string) => ({
         ...record,
-        metric: name,
+        [METRIC_KEY]: name,
         value: record[name],
       })),
     ],
     [],
   );
 
   const [rows, cols] = transposePivot
-    ? [groupbyColumns, ['metric', ...groupbyRows]]
-    : [groupbyRows, ['metric', ...groupbyColumns]];
+    ? [groupbyColumns, [METRIC_KEY, ...groupbyRows]]
+    : [groupbyRows, [METRIC_KEY, ...groupbyColumns]];
+
+  const handleChange = useCallback(
+    (filters: SelectedFiltersType) => {
+      const groupBy = Object.keys(filters);
+      const groupByValues = Object.values(filters);
+      setDataMask({
+        extraFormData: {
+          filters:
+            groupBy.length === 0
+              ? []
+              : groupBy.map(col => {
+                  const val = filters?.[col];
+                  if (val === null || val === undefined)
+                    return {
+                      col,
+                      op: 'IS NULL',
+                    };
+                  return {
+                    col,
+                    op: 'IN',
+                    val: val as (string | number | boolean)[],
+                  };
+                }),
+        },
+        filterState: {
+          value: groupByValues.length ? groupByValues : null,
+        },
+        ownState: {
+          selectedFilters: filters && Object.keys(filters).length ? filters : null,
+        },
```

Comment:
Do we need `ownState`? I think we should be able to get by with just `filterState`. Usually `ownState` is only needed for more complex use cases where the chart's own `buildQuery` needs to be customized based on interaction with the chart.